test/stream.js

```diff
@@ -196,7 +196,7 @@ test('preconnect', async function (t) {
   const socket = new Socket()
   socket.bind(0)
 
-  socket.on('preconnect', (id, address) => {
+  socket.once('preconnect', (id, address) => {
```

Comment:
this was a workaround because the preconnect is triggered twice now depending on timing. i think it's because the ack goes through and triggers it